src/coinsselectionalgorithm.cpp

```diff
@@ -0,0 +1,596 @@
+                    }
+                    //second joinsplit input
+                    joinsplitValue += amounts[windowFrontIndex];
+                    if (joinsplitsOutputAmountIndex < numberOfJoinsplitsOutputsAmounts)
+                    {
+                        if (joinsplitValue >= joinsplitsOutputsAmounts[joinsplitsOutputAmountIndex])
+                        {
+                            changeFromPreviousJoinsplit = joinsplitValue - joinsplitsOutputsAmounts[joinsplitsOutputAmountIndex];
+                            ++joinsplitsOutputAmountIndex;
+                        }
+                        else
+                        {
+                            joinsplitsOutputsAmounts[joinsplitsOutputAmountIndex] -= joinsplitValue;
+                            changeFromPreviousJoinsplit = 0;
+                        }
+                    }
+                    isFirstJoinsplitInput = true;
+                }
+
+                // check upper-limit constraints
+                if (tempTotalSize + (numberOfJoinsplitsOutputsAmounts - joinsplitsOutputAmountIndex) * sizes[0] > availableTotalSize || // first element size is used (but actually all sizes are equal)
+                    tempTotalAmount > targetAmountPlusOffset)
+                {
+                    // if admissible solution still not found, restart with sliding window back index increased by one position
+                    if (!admissibleFound)
+                    {
+                        break;
+                    }
+                    // if admissible solution already found, pop from front of the sliding window only the element just inserted
+                    else
+                    {
+                        tempSelection[windowFrontIndex] = 0;
+                        tempTotalSize -= tempTotalSizeIterationIncrease;
+                        tempTotalAmount -= amounts[windowFrontIndex];
+                        tempTotalSelection -= 1;
+                        bestAdmissibleFound = true;
+                    }                    
+                }
+
+                // check lower-limit constraint
+                if (tempTotalAmount >= targetAmount)
+                {
+                    admissibleFound = true;
+                    // if best admissible solution already found or reached array end, set optimal solution
+                    if (bestAdmissibleFound || windowFrontIndex == 0)
+                    {
+                        optimalTotalSize = tempTotalSize + (numberOfJoinsplitsOutputsAmounts - joinsplitsOutputAmountIndex) * sizes[0]; // first element size is used (but actually all sizes are equal)
+                        optimalTotalAmount = tempTotalAmount;
+                        optimalTotalSelection = tempTotalSelection;
+                        for (int index = 0; index < problemDimension; ++index)
+                        {
+                            optimalSelection[index] = tempSelection[index];
+                        }
```

Comment:
See comments above for copying vector data

Reply:
I agree with your suggestion.
Modified accordingly.